pkg/system/resources.go

```diff
@@ -0,0 +1,69 @@
+package system
+
+import (
+	"os"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+// Values will likely need to be tweaked as we get more usage data. They are currently based
+// on metrics we have collected from internal use. You can override these values by setting
+// the corresponding environment variable.
+var (
+	registryMemoryRequest = "20Mi" // REGISTRY_MEMORY_REQUEST
+	registryMemoryLimit   = "60Mi" // REGISTRY_MEMORY_LIMIT
+	registryCPURequest    = "50m"  // REGISTRY_CPU_REQUEST
+
+	buildkitdMemoryRequest = "70Mi"  // BUILDKITD_MEMORY_REQUEST
+	buildkitdMemoryLimit   = "200Mi" // BUILDKITD_MEMORY_LIMIT
+	buildkitdCPURequest    = "50m"   // BUILDKITD_CPU_REQUEST
+
+	buildkitdServiceMemoryRequest = "50Mi"  // BUILDKITD_SERVICE_MEMORY_REQUEST
+	buildkitdServiceMemoryLimit   = "100Mi" // BUILDKITD_SERVICE_MEMORY_LIMIT
+	buildkitdServiceCPURequest    = "50m"   // BUILDKITD_SERVICE_CPU_REQUEST
+)
+
+func RegistryResources() corev1.ResourceRequirements {
+	return corev1.ResourceRequirements{
+		Requests: corev1.ResourceList{
+			corev1.ResourceMemory: envOrDefault(os.Getenv("REGISTRY_MEMORY_REQUEST"), registryMemoryRequest),
+			corev1.ResourceCPU:    envOrDefault(os.Getenv("REGISTRY_CPU_REQUEST"), registryCPURequest),
+		},
+		Limits: corev1.ResourceList{
+			corev1.ResourceMemory: envOrDefault(os.Getenv("REGISTRY_MEMORY_LIMIT"), registryMemoryLimit),
+		},
+	}
+}
+
+func BuildkitdResources() corev1.ResourceRequirements {
+	return corev1.ResourceRequirements{
+		Requests: corev1.ResourceList{
+			corev1.ResourceMemory: envOrDefault(os.Getenv("BUILDKITD_MEMORY_REQUEST"), buildkitdMemoryRequest),
+			corev1.ResourceCPU:    envOrDefault(os.Getenv("BUILDKITD_CPU_REQUEST"), buildkitdCPURequest),
+		},
+		Limits: corev1.ResourceList{
+			corev1.ResourceMemory: envOrDefault(os.Getenv("BUILDKITD_MEMORY_LIMIT"), buildkitdMemoryLimit),
+		},
+	}
+}
+
+func BuildkitdServiceResources() corev1.ResourceRequirements {
+	return corev1.ResourceRequirements{
+		Requests: corev1.ResourceList{
+			corev1.ResourceMemory: envOrDefault(os.Getenv("BUILDKITD_SERVICE_MEMORY_REQUEST"), buildkitdServiceMemoryRequest),
+			corev1.ResourceCPU:    envOrDefault(os.Getenv("BUILDKITD_SERVICE_CPU_REQUEST"), buildkitdServiceCPURequest),
+		},
+		Limits: corev1.ResourceList{
+			corev1.ResourceMemory: envOrDefault(os.Getenv("BUILDKITD_SERVICE_MEMORY_LIMIT"), buildkitdServiceMemoryLimit),
+		},
+	}
+}
+
+func envOrDefault(env, def string) resource.Quantity {
+	amount := def
+	if env != "" {
+		amount = env
+	}
+	return resource.MustParse(amount)
```

Comment:
Typically, this is frowned upon because, if the environment variable cannot be parsed, then this will panic. That would be OK if this was parsed on startup only, but I don't think that is the case here.

Are you opposed to checking an error and reverting to the default if an error occurs?

Reply:
Nope, not opposed to that at all. I'll get to implementing that.